### PR TITLE
Consignments confirmation screen

### PR DESCRIPTION
**Pull request by orta**
Adds a final step to the consignments process. It will show the submission spinner for 1 second then switch to the fail/success screen.

screen shot 2018-01-05 at 3 54 48 pm
screen shot 2018-01-05 at 3 54 45 pm
screen shot 2018-01-05 at 3 54 39 pm

#skip_new_tests

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  No CHANGELOG added.
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  Missing Test Files:

- `src/lib/Components/Buttons/__tests__/index-tests.tsx`
- `src/lib/Components/UtilityViews/__tests__/RotatingView-tests.tsx`

If these files are supposed to not exist, please update your PR body to include "#skip_new_tests".
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  
<b>🃏 FAIL</b> in <a href="https://github.com/artsy/emission/blob/mc-consignments-confirmationscreen/src/lib/Components/Consignments/Screens/__tests__/Overview-local-storage-tests.tsx">Overview-local-storage-tests.tsx</a>


<li>
<a href='https://github.com/artsy/emission/blob/mc-consignments-confirmationscreen/src/lib/Components/Consignments/Screens/__tests__/Overview-local-storage-tests.tsx#L57'>resets the cache when a final submission is made</a>
<br/>
Expected mock function to have been called with: ["ConsignmentsStoredState"]But it was not called.

<details>
<summary>Full message</summary>
<pre><code>
Error: expect(jest.fn()).toBeCalledWith(expected)

Expected mock function to have been called with:
  ["ConsignmentsStoredState"]
But it was not called.
    at _callee$ (/home/travis/build/artsy/emission/src/lib/Components/Consignments/Screens/__tests__/Overview-local-storage-tests.tsx:57:35)
    at tryCatch (/home/travis/build/artsy/emission/node_modules/react-native/node_modules/regenerator-runtime/runtime.js:63:40)
    at Generator.invoke [as _invoke] (/home/travis/build/artsy/emission/node_modules/react-native/node_modules/regenerator-runtime/runtime.js:337:22)
    at Generator.prototype.(anonymous function) [as next] (/home/travis/build/artsy/emission/node_modules/react-native/node_modules/regenerator-runtime/runtime.js:96:21)
    at fulfilled (/home/travis/build/artsy/emission/src/lib/Components/Consignments/Screens/__tests__/Overview-local-storage-tests.tsx:10:54)
    at tryCallOne (/home/travis/build/artsy/emission/node_modules/promise/lib/core.js:37:12)
    at /home/travis/build/artsy/emission/node_modules/promise/lib/core.js:123:15
    at flush (/home/travis/build/artsy/emission/node_modules/asap/raw.js:50:29)
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
    at process._tickCallback (internal/process/next_tick.js:180:9)
</code></pre>
</details>
</li>
<br/>


  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  
<b>🃏 FAIL</b> in <a href="https://github.com/artsy/emission/blob/mc-consignments-confirmationscreen/src/lib/Components/Consignments/Screens/__tests__/Overview-analytics-tests.tsx">Overview-analytics-tests.tsx</a>


<li>
<a href='https://github.com/artsy/emission/blob/mc-consignments-confirmationscreen/src/lib/Components/Consignments/Screens/__tests__/Overview-analytics-tests.tsx#L34'>calls the draft created event</a>
<br/>


<details>
<summary>Full message</summary>
<pre><code>
TypeError: this.props.navigator.push is not a function
    at Info.submissionDraftSubmitted (/home/travis/build/artsy/emission/src/lib/Components/Consignments/Screens/Overview.tsx:162:16)
    at Info.decorateClassMember (/home/travis/build/artsy/emission/node_modules/react-tracking/build/trackEventMethodDecorator.js:29:34)
    at Object.<anonymous> (/home/travis/build/artsy/emission/src/lib/Components/Consignments/Screens/__tests__/Overview-analytics-tests.tsx:34:3)
    at Object.asyncFn (/home/travis/build/artsy/emission/node_modules/jest-jasmine2/build/jasmine_async.js:124:345)
    at resolve (/home/travis/build/artsy/emission/node_modules/jest-jasmine2/build/queue_runner.js:46:12)
    at tryCallTwo (/home/travis/build/artsy/emission/node_modules/promise/lib/core.js:45:5)
    at doResolve (/home/travis/build/artsy/emission/node_modules/promise/lib/core.js:200:13)
    at new Promise (/home/travis/build/artsy/emission/node_modules/promise/lib/core.js:66:3)
    at mapper (/home/travis/build/artsy/emission/node_modules/jest-jasmine2/build/queue_runner.js:34:499)
    at promise.then (/home/travis/build/artsy/emission/node_modules/jest-jasmine2/build/queue_runner.js:74:39)
</code></pre>
</details>
</li>
<br/>


  </td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by orta** [comment]
Oh yeah nice, I forgot that we get jest logs in Danger

**Comment by ashfurrow** [comment]
Overall looks good, though I'm a bit nervous about this 1-second timer to check for submission status. Reading the code, it appears that if the request takes longer than a second to return (which isn't too unreasonable given sufficiently terrible cellular reception), then the consignment submission is displayed as having failed? Would it be possible to poll for the response, or even tie the UI to the request directly?

**Comment by orta** [comment]
> Would it be possible to poll for the response, or even tie the UI to the request directly?

Yep, done.

File: src/lib/Components/Consignments/Screens/__tests__/Confirmation-tests.tsx [hunk 1]
```diff
@@ -56,9 +56,25 @@ describe("callbacks", () => {
     const confirmation = new Confirmation({ ...emptyProps, submissionRequestValidationCheck })
     confirmation.setState = jest.fn()
 
-    jest.runAllTimers()
+    jest.runOnlyPendingTimers()
     expect(confirmation.setState).toHaveBeenCalledWith({ submissionState: "FailedSubmission" })
   })
+
+  it.only("getting undefined back sets will make it run the check a second time", () => {
```
Comment (ashfurrow) [inline src/lib/Components/Consignments/Screens/__tests__/Confirmation-tests.tsx]: `.only` left in?

Comment (orta) [reply src/lib/Components/Consignments/Screens/__tests__/Confirmation-tests.tsx]: Nice spot

**Pull request merged**